tests/test_core/test_set_params.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+
+from etna.models import CatBoostMultiSegmentModel
+from etna.pipeline import Pipeline
+from etna.transforms import AddConstTransform
+
+
+def test_base_mixin_set_params_changes_params_estimator():
+    catboost_model = CatBoostMultiSegmentModel(iterations=1000, depth=10)
+    catboost_model = catboost_model.set_params(**{"learning_rate": 1e-3, "depth": 8})
+    expected_dict = {
+        "_target_": "etna.models.catboost.CatBoostMultiSegmentModel",
+        "iterations": 1000,
+        "depth": 8,
+        "learning_rate": 1e-3,
+        "logging_level": "Silent",
+        "kwargs": {},
+    }
+    obtained_dict = catboost_model.to_dict()
+    assert obtained_dict == expected_dict
+
+
+def test_base_mixin_set_params_changes_params_pipeline():
+    pipeline = Pipeline(model=CatBoostMultiSegmentModel(iterations=1000, depth=10), transforms=(), horizon=5)
+    pipeline = pipeline.set_params(
+        **{"model.learning_rate": 1e-3, "model.depth": 8, "transforms": AddConstTransform("column", 1)}
+    )
+    expected_dict = {
+        "_target_": "etna.pipeline.pipeline.Pipeline",
+        "horizon": 5,
+        "model": {
+            "_target_": "etna.models.catboost.CatBoostMultiSegmentModel",
+            "depth": 8,
+            "iterations": 1000,
+            "kwargs": {},
+            "learning_rate": 0.001,
+            "logging_level": "Silent",
+        },
+        "transforms": {
+            "_target_": "etna.transforms.math.add_constant.AddConstTransform",
+            "in_column": "column",
+            "inplace": True,
+            "value": 1,
+        },
+    }
+    obtained_dict = pipeline.to_dict()
+    assert obtained_dict == expected_dict
+
+
+def test_base_mixin_set_params_doesnt_change_params_inplace_estimator():
+    catboost_model = CatBoostMultiSegmentModel(iterations=1000, depth=10)
+    catboost_model.set_params(**{"learning_rate": 1e-3, "depth": 8})
+    expected_dict = {
+        "_target_": "etna.models.catboost.CatBoostMultiSegmentModel",
+        "iterations": 1000,
+        "depth": 10,
+        "logging_level": "Silent",
+        "kwargs": {},
+    }
+    obtained_dict = catboost_model.to_dict()
+    assert obtained_dict == expected_dict
+
+
+def test_base_mixin_set_params_doesnt_change_params_inplace_pipeline():
+    pipeline = Pipeline(model=CatBoostMultiSegmentModel(iterations=1000, depth=10), transforms=(), horizon=5)
+    pipeline.set_params(**{"model.learning_rate": 1e-3, "model.depth": 8, "transforms": AddConstTransform("column", 1)})
```

Comment:
Parameter `transforms` should be sequence, let's wrap `AddConstTransform` in a list.

Reply:
What about this?

Reply:
resolved

Reply:
I see `"transforms": AddConstTransform("column", 1)`, it is not wrapped in a list like: `"transforms": [AddConstTransform("column", 1)]`

Reply:
Oh, my bad.